libp2p/builders.nim

```diff
@@ -152,9 +151,8 @@ proc withNameResolver*(b: SwitchBuilder, nameResolver: NameResolver): SwitchBuil
   b.nameResolver = nameResolver
   b
 
-proc withRelayTransport*(b: SwitchBuilder, canHop: bool): SwitchBuilder =
-  b.isCircuitRelay = true
-  b.circuitRelayCanHop = canHop
+proc withCircuitRelay*(b: SwitchBuilder, r: Relay): SwitchBuilder =
```

Comment:
If we do as my last comment suggest, we could do `r = Relay.new()`, since for relay only setup, the user doesn't need the relay